src/Pages/Search/Search.tsx

```diff
@@ -1,35 +1,95 @@
 import { getPokemonsData } from "../../app/getPokemonData";
 import InputBar from "../../components/InputBar/InputBar";
-
-const Search:React.FC = () => {
+import Loader from "../../components/Loader/Loader";
+import { increaseOffset } from "../../app/pokemonListSlice";
+const Search: React.FC = () => {
+  // const offsetRef = useRef<number>(0);//redux the offset
+  // const limitRef = useRef<number>(20);// redux the limit
+  const targetRef = useRef<HTMLDivElement>(null);
   const dispatch = useAppDispatch();
-  // data fetching from api and storing it to the store
-   
-  const { allPokemon, randomPokemon } = useAppSelector(
+  const [scrollPosition, setScrollPosition] = useState("");
+  const { allPokemon, randomPokemon, offset, limit } = useAppSelector(
     ({ pokemon }) => pokemon
   );
+
+  const previousOffsetRef = useRef<number>(offset);
   useEffect(() => {
-    dispatch(getInitialPokemonData());
+    previousOffsetRef.current = offset;
+  }, [offset]);
+  const observer = useRef<IntersectionObserver | null>(null);
+
+  useEffect(() => {
+    observer.current = new IntersectionObserver((entries) => {
+      if (entries[0].isIntersecting) {
+        console.log("offSet is " + previousOffsetRef.current);
+        dispatch(getInitialPokemonData(previousOffsetRef.current, limit));
+        dispatch(increaseOffset());
+      }
+    });
+
+    return () => {
+      if (observer.current) {
+        observer.current.disconnect();
+      }
+    };
   }, [dispatch]);
+
+  useEffect(() => {
+    if (targetRef.current && observer.current) {
+      observer.current.observe(targetRef.current);
+    }
+  }, [observer]);
+
   useEffect(() => {
     if (allPokemon) {
-      const clonedPokemon = [...allPokemon];
-      const randomPokemons = clonedPokemon
-        .sort(() => Math.random() - Math.random())
-        .slice(0, 20);
-      dispatch(getPokemonsData(randomPokemons));
+      const clonedPokemon = [...allPokemon]; //X
+      const randomPokemons = allPokemon;
+
+      dispatch(getPokemonsData(randomPokemons)); //x
     }
   }, [allPokemon, dispatch]);
+  /// __________________auto scroll to the last scroll position ____________________________________//
+  const scrollableRef = useRef<HTMLDivElement>(null);
+
+  const handleScroll = () => {
+    console.log("handlescroll is called ");
+    if (scrollableRef.current) {
+      const { scrollTop, scrollLeft } = scrollableRef.current;
+      console.log("Scroll Top:", scrollTop);
+      console.log("Scroll Left:", scrollLeft);
+    }
+  };
+
+  useEffect(() => {
+    if (scrollableRef.current) {
+      scrollableRef.current.addEventListener("scroll", handleScroll);
+    }
+
+    return () => {
+      if (scrollableRef.current) {
+        scrollableRef.current.removeEventListener("scroll", handleScroll);
+      }
+    };
+  }, []);
+
+  /// ________________________________________________//
 
   return (
-    <div className=" flex flex-wrap">
+    <div className=" flex flex-wrap ">
```

Comment:
don't think flex-wrap is required.

```suggestion
    <div className=" flex flex-col">
```

Reply:
changed it to flex flex-col